server/Middleware/auth.js

```diff
@@ -0,0 +1,21 @@
+const jwt = require("jsonwebtoken");
+
+const config = process.env;
+
+const verifyToken = (req, res, next) => {
+  const token =
+    req.body.authorization || req.query.authorization || req.headers["authorization"];
+
+  if (!token) {
+    return res.status(403).send("A token is required for authentication");
+  }
+  try {
+    const decoded = jwt.verify(token, config.SECRET_KEY);

```

Comment:
Any pros and cons on using JWTs as auth?

Reply:
using JWTs implies foregoing revocation which leads to :
- logout not functioning as it should because the system will only really logout when the token expires
- blocking users doesn't really block them immediately. The user will continue having access until their token expires
- possibility of having stale data in case of downgrade/upgrade of user permissions.

The advantages of JWT include:
- no cookies required hence mobile friendly
- It is stateless since it has no session to manage
- reduces network round trip time

Reply:
@mutebironald any workarounds for the cons of JWT?

Reply:
you can use OAuth2 as a workaround.

Reply:
@mutebironald that is not really an workaround. But you can use redis DB and have an extra check on the tokens if they are valid or not. That way you do have a way to deny certain tokens acces. But then you can just use cookies. But it is a workaround if you use the other capabilities of the jwt token, and have more attrbiutes inside it